app/api/server/v1/teams.ts

```diff
@@ -189,15 +189,24 @@ API.v1.addRoute('teams.listRoomsOfUser', { authRequired: true }, {
 API.v1.addRoute('teams.members', { authRequired: true }, {
 	get() {
 		const { offset, count } = this.getPaginationItems();
-		const { teamId, teamName } = this.queryParams;
-		const { query } = this.parseJsonQuery();
+		const { teamId, teamName, status, username, name } = this.queryParams;
+
+		if (status && !Array.isArray(status)) {
+			throw new Meteor.Error('error-status-param-not-an-array', 'The parameter "status" should be an array of strings');
+		}
 
 		const team = teamId ? Promise.await(Team.getOneById(teamId)) : Promise.await(Team.getOneByName(teamName));
 		if (!team) {
 			return API.v1.failure('team-does-not-exist');
 		}
 		const canSeeAllMembers = hasPermission(this.userId, 'view-all-teams', team.roomId);
 
+		const query = {
+			username,
```

Comment:
Is it expected that username/name are not RegExp here?

Reply:
Yes, so I should use `check` to assure this. Right?

Reply:
I don't think so 🤔 I was thinking that most username/name filters support regexes for searching, but if that's not needed here, then it's fine

Reply:
Not sure, but I think that supporting regexes would require some extra work (such as in the `getUsersOfRoom` method). @sampaiodiego , should this endpoint support RegExp?

Reply:
`ping` @sampaiodiego

Reply:
oops, sry, I forgot to answer the last mention.. 

so I think the main idea is to be able to use those `*.members` endpoints wherever `getUsersOfRooms` is being currently used:
https://github.com/RocketChat/Rocket.Chat/blob/f8171f464ed8a7487795651767695fb33a1c709e/server/methods/getUsersOfRoom.js#L32

in that sense it should accept a regex as well..

Reply:
In that sense, i guess the `im.members` PR (#21471) should be updated too to support a regex 🤔